Add readme.md in demo-forms dir: how to load forms into CHT #6728 (#6729)
medic/cht-core: demo-forms/readme.md

# Demo Forms

## Intro

These forms can be used to quickly test specific XForm features. For example, if you load the `geopoint` form, you can test the Enketo Geopoint feature in an isolated way without having to worry if other form fields might be causing issues.

## Loading

The forms can be loaded via the web admin GUI or through the command line using `medic-conf`.

### Admin Web GUI

> 1. Log into your CHT instance as an admin user. Go to the "App Management" section at `/admin/#/forms`

Author: Oh - yeah - I'm totally open to doing it differently. I'm trying to acknowledge that we don't know the URL of the CHT instance so I specify the URI as well as the section name. Likely it's `localhost` in developer environment. I'm open to changing this doc if there's a more standard practice.

Add a link to docs per you recommendation - thanks!

# Demo Forms

## Intro

> These forms can be used to quickly test specific XForm features. For example, if you load the `geopoint` form, you can test the Enketo Geopoint feature in an isolated way without having to worry about whether or not other form fields might be causing issues.

Reviewer: ```suggestion
These forms can be used to quickly test specific XForm features. For example, if you load the `geopoint` form, you can test the Enketo Geopoint feature in an isolated way without having to worry if other form fields might be causing issues.
```

## Loading

> The forms can either be loaded in via the web admin GUI or through command line using `medic-conf`. 

Reviewer: ```suggestion
The forms can be loaded via the web admin GUI or through command line using `medic-conf`.
```

### Admin Web GUI

1. Log into your CHT instance as an admin user. Go to the "App Management" section at `/admin/#/forms`
1. At the top of the page under "Upload app forms", under "XML" choose "Browse..." and find the XML of the form you want to add in the `demo-forms` directory (eg. `geopoint.xml`).

> 1. Under "Meta" choose "Browser..." again and select `app-form.properties.json`

Reviewer: ```suggestion
1. Under "Meta" choose "Browse..." again and select `app-form.properties.json`
```

Do you want "Browse" or "Browser" here?

Author: Indeed - nice catch!

1. Submit the new form by clicking "Upload".  You should see a brief spinner and the "Upload app forms" section will reset to being empty as you first found it.
1. Navigate back to the reports page of the app at `/#/reports/` and under the lower left "Submit report", you should now see your new form (e.g. "Geopoint Demo Form").

### medic conf

> 1. Copy the form you want to upload from the  `demo-forms` to the `config/default/forms/app` directory. For the `geopoint.xml` form, if you were in the root of this repo, that would be: `cp demo-forms/geopoint.xml config/default/forms/app/`

Reviewer: ```suggestion
1. Copy the form you want to upload from the `demo-forms` to the `config/default/forms/app` directory. For the `geopoint.xml` form, if you were in the root of this repo, that would be: `cp demo-forms/geopoint.xml config/default/forms/app/`
```

Just removing an extra space here...

1. Change to the default config `cd config/default`

> 1. Run `medic-conf` specifying the `upload-app-forms` command along with the specific form you want to upload. This example shows uploading the `geopoint` form.  Note that the trailing `.xml` is not included: `medic-conf --local upload-app-forms -- geopoint`

Reviewer: ```suggestion
1. Run `medic-conf` specifying the `upload-app-forms` command along with the specific form you want to upload. This example uses the `geopoint` form.  Note that the trailing `.xml` is not included: `medic-conf --local upload-app-forms -- geopoint`
```
